bold-linkage/main.py

```diff
@@ -0,0 +1,224 @@
+
+    # Get one or more specimen IDs. Considering that this will be a GET request this can't
+    # be enormous - but perhaps 100 IDs is fine as a batch size.
+    identifiers = list(
+        map(
+            lambda identifier: identifier.get("???:identifierValue"),
+            specimen_data.get("identifiers"),
+        )
+    )
+
+    # BOLD's API has a concept of 'scope' (here: 'ids') and 'subscope' (here: 'sampleid'),
+    # where the value is the third part of a triple. All triples are joined with commas.
+    query_value = ",".join([f"ids:sampleid:{id}" for id in identifiers])
+
+    # Compose the URL and define the accept header
+    headers = {"Accept": "application/json"}
+    query_string = f"{query_endpoint}?query={query_value}&extent=full"
+
+    # Do the request
+    response = requests.get(
+        query_string,
+        headers=headers,
+        auth=HTTPBasicAuth(os.environ.get("API_USER"), os.environ.get("API_PASSWORD")),
+    )
+    response.raise_for_status()  # Raises an HTTPError if the status is 4xx, 5xx
+
+    # The response is a token from which a URL needs to be composed whose dereferencing
+    # results in a set of records
+    query_id = response.json()["query_id"]
+
+    # Compose the URL for getting results. It is possible to page through results, so
+    # you can specify how many you want per page (`length`) and the zero-based offset
+    # assuming no more than 25 results per specimen are returned
+    docs_endpoint = f"https://boldsystems.eu/api/documents/{query_id}?length=25&start=0"
+
+    # Fetch the documents
+    response = requests.get(
+        docs_endpoint,
+        headers=headers,
+        auth=HTTPBasicAuth(os.environ.get("API_USER"), os.environ.get("API_PASSWORD")),
+    )
+    response.raise_for_status()  # Ensure the request was successful
+
+    # Parse the response to get the records
+    records = response.json()["data"]
+
+    return list(
+        map(
+            lambda record: {
+                "queryString": query_string,
+                "processid": record["processid"],
+            },
+            records,
+        )
+    )
+
+
+def run_local(example: str) -> None:
+    """
+    Run the script locally. Can be called by replacing the kafka call with this  a method call in the main method.
+    Will call the DiSSCo API to retrieve the specimen data.
+    A record ID will be created but can only be used for testing.
+    :param example: The full URL of the Digital Specimen to the API (for example
+    https://dev.dissco.tech/api/v1/specimens/TEST/S0P-JMS-V4Q
+    :return: Return nothing but will log the result
+    """
+    response = requests.get(example)
+    specimen = json.loads(response.content)["data"]
+    specimen_data = specimen["attributes"]["digitalSpecimen"]
+    result = run_api_call(specimen_data)
+    mas_job_record = map_to_mas_job_record(specimen_data, result, str(uuid.uuid4()))
+    logging.info("Created annotations: " + str(mas_job_record))
```

Comment:
`logging.info("Created annotations: " + json.dumps(mas_job_record, indent=2))` for pretty print?